core/transaction.go

```diff
@@ -220,36 +279,33 @@ func declareTransactionHash(d *DeclareTransaction, network utils.Network) (*felt
 	return nil, errors.New("invalid transaction version")
 }
 
-type L1HandlerTransaction struct {
-	// todo
-}
-
-func (h *L1HandlerTransaction) Hash(network utils.Network) (*felt.Felt, error) {
-	panic("not implemented")
+func l1HandlerTransactionHash(l *L1HandlerTransaction, network utils.Network) (*felt.Felt, error) {
+	// TODO: implement me
+	panic("implement me")
 }
 
-type DeployAccountTransaction struct {
-	// todo
-}
-
-func (d *DeployAccountTransaction) Hash(network utils.Network) (*felt.Felt, error) {
-	panic("not implemented")
+func deployAccountTransactionHash(d *DeployAccountTransaction,
+	network utils.Network,
+) (*felt.Felt, error) {
+	// TODO: implement me
+	panic("implement me")
 }
 
 const commitmentTrieHeight uint = 64
 
 // TransactionCommitment is the root of a height 64 binary Merkle Patricia tree of the
 // transaction hashes and signatures in a block.
-func TransactionCommitment(receipts []*TransactionReceipt) (*felt.Felt, error) {
+func TransactionCommitment(transactions []Transaction) (*felt.Felt, error) {
 	var transactionCommitment *felt.Felt
 	return transactionCommitment, trie.RunOnTempTrie(commitmentTrieHeight, func(trie *trie.Trie) error {
-		for i, receipt := range receipts {
-			signaturesHash := crypto.PedersenArray()
-			if receipt.Type == Invoke {
-				signaturesHash = crypto.PedersenArray(receipt.Signatures...)
+		for i, transaction := range transactions {
+			signatureHash := crypto.PedersenArray()
+			if _, ok := transaction.(*InvokeTransaction); ok {
```

Comment:
this if is not needed anymore, right? signature() will return an empty array for non-invoke txs.

Reply:
Some other transaction types actually have a Signature (e.g. Declare)

Reply:
then if must go